session_ii/javascript-fundamentals/session2/homework/mpricu/function.js

```diff
@@ -0,0 +1,79 @@
+
+const pro1 = new Promise((resolve, reject) => {
+    // setTimeout(() => resolve(1), 1000);
```

Comment:
If we uncomment this line and use a timeout value like 3000 and comment out the next one, the result will not be the expected one. The result should be a fulfillment because we have at least one promise that was fulfilled.

Reply:
The function allReject should have a correlated value with the longest settle value for any promise. If you modify to allReject(5) it should work.

Reply:
but how could you know beforehand how long you have to wait? When you're subscribing to something the subject/producer decides when the value is ready.

Reply:
No, you do not know, that's why the original message for the allReject promise was : "Timeout !". That promise will be settled after a given amount of time you decided. In real life you will not wait forever for the server response.
I said 5 seconds because you have chosen 3 seconds  for first promise to be resolved. In rel life you do not know that value is 3 seconds. So if I know that value I can adjust my function. Otherwise you put the maximum time you can afford to wait.

Reply:
In real life scenarios you might use a timeout, but the requirements for this exercise don't allow this here. :-)

Reply:
You see, that's why I asked for the real use cases where the promises are applied...and that;s why I put the message "ihatepromisesbecauseilovefacts".
In real life you have to get data from an api endpoints or multiple endpoints and you need to know which solution to apply and when.
And it is not too late to make this summary of solutions....:)....will be very useful for everybody.